api/python/provisioner/config.py

```diff
@@ -378,10 +378,32 @@ class Checks(Enum):
     CLUSTER_STATUS = "cluster_status"
     LOGS_ARE_GOOD = "logs_are_good"
     PASSWORDLESS_SSH_ACCESS = "passwordless_ssh_access"
+    STORAGE_LVMS = "storage_lvms"
+    STORAGE_LUNS = "storage_luns"
+    MGMT_VIP = "mgmt_vip"
+    HOSTNAMES = "hostnames"
+    POST_CHECKS = "post_checks"
+    REPLACENODE_CHECKS = "replacenode_checks"
 
 
 # Set of supported validations/checks
 CHECKS = {
+    Checks.NETWORK.value,
+    Checks.CONNECTIVITY.value,
+    Checks.BMC_ACCESSIBILITY.value,
+    Checks.COMMUNICABILITY.value,
+    Checks.CLUSTER_STATUS.value,
+    Checks.LOGS_ARE_GOOD.value,
+    Checks.PASSWORDLESS_SSH_ACCESS.value,
+    Checks.STORAGE_LVMS.value,
+    Checks.STORAGE_LUNS.value,
+    Checks.MGMT_VIP.value,
+    Checks.HOSTNAMES.value,
+    Checks.POST_CHECKS.value,
```

Comment:
I don't understand the design:
- granular checks are combined with group checks
- swupdate (as a group check) is not listed though
- sw update checks are triggered for ALL value but list for ALL != list for sw update

Reply:
We can also consider the creation of different classes/types for granular checks and group checks. This will allow us to use `type/isinstance` python API, for example.

Reply:
Simplified design by making 2 different classes for checks:

``` class Checks(Enum):

    """Enumeration for available checks/validations"""

    NETWORK = "network"
    CONNECTIVITY = "connectivity"
    BMC_ACCESSIBILITY = "bmc_accessibility"
    COMMUNICABILITY = "communicability"
    CLUSTER_STATUS = "cluster_status"
    LOGS_ARE_GOOD = "logs_are_good"
    PASSWORDLESS_SSH_ACCESS = "passwordless_ssh_access"
    STORAGE_LVMS = "storage_lvms"
    STORAGE_LUNS = "storage_luns"
    MGMT_VIP = "mgmt_vip"
    HOSTNAMES = "hostnames"


class GroupChecks(Enum):

    """Enum for group checks"""
    ALL = "all"
    POST_CHECKS = "post_checks"
    REPLACENODE_CHECKS = "replacenode_checks"
    SWUPDATE_CHECKS = "swupdate_checks"
```